redis_commands.c

```diff
@@ -1355,9 +1347,14 @@ int redis_set_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
                 set_type = Z_STRVAL_P(v);
             }
         } ZEND_HASH_FOREACH_END();
-    } else if (z_opts && Z_TYPE_P(z_opts) == IS_LONG) {
-        expire = Z_LVAL_P(z_opts);
+    } else if (z_opts && Z_TYPE_P(z_opts) != IS_NULL) {
+        if (Z_TYPE_P(z_opts) != IS_LONG && Z_TYPE_P(z_opts) != IS_STRING) {
```

Comment:
Don't we need add check for `IS_DOUBLE` and additional check `is_numeric_string` for `IS_STRING`?

Reply:
Good point, we should accept `IS_DOUBLE` `zval`s as well.

I'm not sure how much `is_numeric_string` gets us because I believe non-strings will resolve to `0` which will fail before sending the command because I verify that the `TTL` is greater than zero. 

I suppose it would let us be even more specific about the error but I'm not sure it's worth the extra logic.